### DISCUSSION TITLE
HomeKit automation issue?

### DISCUSSION THREAD

**Discussion by sahafeez** [kuestess/homebridge-platform-insteonlocal#226] [General]
I am wondering how many people are using the Home app for setting up automation for their Insteon controlled devices and does it work reliably? My hub is one of 3 AppleTVs. They are all wired connections. There seems to be no consistence if the automation works or not.

If anyone has had success I would love to hear about your setup!

**Comment by 1004PTLN** [comment]
Hope this is somewhat helpful. 

I have yet to move to @kuestess tool although as you've found he's amazingly responsive.  He helped me when I was trying to repurpose my old Insteon HomeKit hub.  

Long story somewhat short.  I run my Insteon set up through Indigo Domotics software connected through my PLM.  I paid for a year of their service and will likely explore @kuestess  plugin again when that runs out.  

Regardless, your question pertains to HomeKit and my experience has been rock solid by constructing my automations solely through HomeKit and not Indigo.  Which is a bit counterintuitive given how robust Indigo automations are. I wanted to be as simplified as possible and on HomeKit as much as feasible.  

As such, I only use the Indigo software for the HomeKit plugin...which I have to imagine is much like the one  @kuestess wrote. Another issue is that their current HomeKit plugin is not being actively supported, however it looks like another user may have recently begun to fork another option.  

Not sure how big or complex your setup is, but for reference I have about 60 Insteon devices (mostly dual band dimmers, a few door sensors and a few outdoor modules and lamplincs). 

On HomeKit I've got at least a dozen straightforward lighting automations that work virtually without exception.  

I also have some motion related automations for various closet lights switched on with door open and off with HomeKit built in timers that are controlled by the open/close sensors i mentioned.  

The other benefit is being able to use apple shortcuts for more complex automations through their scripting process.  

In any event, good luck.  We're all charting new waters.  I'm glad I religiously stayed away from Insteons cloud service for sure.  

Last quick point, if you aren't already you should monitor Universal Devices (UDI) forum.  They make the ISY.  It looks like their owners have bid on some Insteon IP including the PLM, which as we all know, is the major Achilles heel for us.

**Comment by sahafeez** [comment]
@1004PTLN Thanks. I have over 85 devices configured and 30 still to add. 

From what I understand from what you wrote you have:
 HK<>Indigo<>PLM as the communication path and it works well.

My setup would be:
 HK<>InsteonLocal<>PLM and I am having issues. 

I am sure @kuestess will be able to sort it if I can do a bit of digging first to point him in the right direction. I like to make one change at a time, so I just switched the mDNS to Ciao, and will test.

**Reply by 1004PTLN** [reply]
That is correct.  

On a related note. There is a plugin on Indigo called "Survey" that does just that. It sends a ping to each Insteon device on the network and provides quite a bit of useful data, such as Insteon address, model type, number of pings, ping time, hop heat map. I used this when setting up my network to locate a few misbehaving switches that for whatever reason were causing noise/congestion/multiple hops on the lines. 

@kuestess is any utility like this available in the insteonlocal universe?

It's a great Insteon network troubleshooting utility.

**Reply by sahafeez** [reply]
The old unsupported houselinc software allowed you to do this. I hooked up my PLM to it and ran the test. 100% across the board.

**Reply by sahafeez** [reply]
@1004PTLN so I looked at Indigo and even downloaded a trial, but it does not work on the latest version of MacOS and every system I have is on the latest. Oh well.

**Reply by 1004PTLN** [reply]
Yes, another reason I'll likely move as well. Currently have it on an old headless Mac but that too will need to change at some point.

**Comment by kuestess** [comment]
@sahafeez I’ve got a handful of Apple TV’s and an iPad that are HomeKit hubs, and my automations are reliable - can’t remember a time (that wasn’t self-induced) that they didn’t work.  @1004PTLN My goal was to make the ui similar to HouseLinc, so certainly can add something like the ‘Survey’ feature.  It may be a bit before I can get to something like that as much of my time currently is focused on issues/support given the influx of new users and other changes/updates planned prior.

**Reply by sahafeez** [reply]
@kuestess I switched to ciao for mDNS and so far things seems to work better. I will let you know in 17 minutes if the scene triggers off correctly. :)

**Reply by sahafeez** [reply]
@kuestess everything turned off. happy, happy, joy, joy!

**Reply by kuestess** [reply]
Awesome - glad it worked!  Now that you mention it, I do remember seeing that Ciao was preferred to Bonjour but I don’t remember where (maybe on homebridge repo).

**Comment by 1004PTLN** [comment]
@kuestess completely understandable!  Given the current situation I'm sure you've become quite popular and the work load has grown commensurately!  I hope you keep things going strong.

**Reply by 1004PTLN** [reply]
@kuestess , circling back on this thread as it's time for me to finally move off of Indigo. Is there anything I should be aware of when making the switch over with respect to any conflicts, plm or otherwise.  Besides the main write up you have done, is there other documentation you would point me to.  I don't want to waste your time, but also want to make the move without major hiccups.  No rush as I plan to do this in the coming month or two.  
As a quick refresher, my set up is my plm connects to a Mac server running Indigo.  I have the HomeKit bridge running in Indigo, no automations are done in Indigo, it's simply there for the HomeKit connection. All automations are done inside HomeKit. I currently use homebridge on a pc server and have a number of plugins working successfully there.  I'd like to migrate all of my Insteon needs to the homebridge server with your system. Appreciate any advice/insight.  Thanks in advance.
